portfolio/src/main/java/com/google/sps/servlets/VoteServlet.java

```diff
@@ -0,0 +1,73 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/vote-results")
+public class VoteServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String poll = request.getParameter("poll");
+
+    Filter pollFilter = new FilterPredicate("poll", FilterOperator.EQUAL, poll);
+    Query query = new Query("Ballot").setFilter(pollFilter);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    List<Entity> results = datastore.prepare(query).asList(FetchOptions.Builder.withDefaults());
+
+    // Count the number of votes for each category to minimize JSON reponse.
+    Map<String, Long> categoryVotes = 
+        results.stream()
+               .map(e -> (String) e.getProperty("vote"))
+               .collect(Collectors.groupingBy(str -> str, Collectors.counting()));
```

Comment:
Nice! Good use of the API.

If you want to go even further, `str -> str` can be Function::identity (https://docs.oracle.com/javase/8/docs/api/java/util/function/Function.html#identity--)

Reply:
Very helpful! I've optimized the stream, greatly appreciate the input.